tests/test_ble_interface.py

```
@@ -0,0 +1,1418 @@
+    # Create a mock queue with a runnable that raises exceptions
+    class ExceptionRunnable:
+        """Mock runnable that raises ValueError when called."""
+
+        def __call__(self):
+            """
+            A callable test stub that always raises ValueError to simulate a failing callback.
+
+            Raises:
+                ValueError: Always raised to indicate the mock callback failed during execution.
+            """
+            raise ValueError("Callback execution failed")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)